System/DataModule.cpp

```diff
@@ -316,23 +316,45 @@ namespace RTE {
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-	bool DataModule::GetAllOfGroup(std::list<Entity *> &entityList, const std::string &group, const std::string &type) {
-		if (group.empty()) {
+
+	bool DataModule::GetAllOfOrNotOfGroups(std::list<Entity *> &entityList, const std::string &type, const std::variant<const std::string, const std::vector<std::string>> &groups, bool excludeGroups) {
+		std::vector<std::string> groupsList;
+		if (const std::string *group = std::get_if<const std::string>(&groups)) {
+			if (!group->empty()) { groupsList.emplace_back(*group); }
+		} else {
+			groupsList = std::get<const std::vector<std::string>>(groups);
+		}
+		if (groupsList.empty()) {
 			return false;
 		}
 
 		bool foundAny = false;
 
-		// Find either the Entity typelist that contains all entities in this DataModule, or the specific class' typelist (which will get all derived classes too)
-		std::map<std::string, std::list<std::pair<std::string, Entity *>>>::iterator classItr = m_TypeMap.find((type.empty() || type == "All") ? "Entity" : type);
-
-		if (classItr != m_TypeMap.end()) {
+		// Find either the Entity typelist that contains all entities in this DataModule, or the specific class' typelist (which will get all derived classes too).
+		if (std::map<std::string, std::list<std::pair<std::string, Entity *>>>::iterator classItr = m_TypeMap.find((type.empty() || type == "All") ? "Entity" : type); classItr != m_TypeMap.end()) {
```

Comment:
Hmm, just realized these use `maps` instead of `unordered_maps`
Feel like creeping up the scope and changing that?

Also, you can probably auto this longass line. If so, I'd say this is a case where it's the right choice for readability

Also also, wtf is with the bit after the semicolon here `m_TypeMap.find((type.empty() || type == "All") ? "Entity" : type); classItr != m_TypeMap.end()`? I guess it's legal, but why u add it?